mon-pix/app/templates/certifications/results.hbs

```diff
@@ -1,2 +1,2 @@
 {{title pageTitle}}
-<CertificationResultsPage @certificationNumber={{@model}} />
+<CertificationResultsPage @class="certification-results-page" @certificationNumber={{@model}} />
```

Comment:
On envoie cette classe css au composant, mais le composant ne l'utilise pas.
Ce ne serait pas plutôt une encapsulation `<div class='certification-results-page'></div>` qu'il faudrait ajouter dans `certification-results-page.hbs`? Ou encore `@classNames('tutorial-panel')` sur `certification-results-page.js`
Ou bien est-ce que la "magie" s'opère toute seule avec `@class` ?

Reply:
Il me semble que `@class` est équivalent à une encapsulation de dans une div justement. `@class` ne constitue pas un paramètre.
`@classNames('tutorial-panel')` dans `certification-results-page.js` est une approche non recommandée par Ember qui reste tolérée le temps de la migration

Reply:
Sauf si j'ai raté un truc, `@class` est bien un paramètre du composant et se retrouvera dans `this.args`. @celineung a raison, si le composant n'utilise pas `this.args.class` ça n'aura pas d'effet. À distinguer de l'attribut `class` qui serait utilisable avec `...attributes` dans le template (avec un traitement spécifique pour `class` d'ailleurs). La réf : https://guides.emberjs.com/release/components/component-arguments-and-html-attributes/#toc_html-attributes 

De fait, le nouveau composant (à gauche sur la photo) ne génère plus de `div` englobante et il n'y a donc pas de place pour l'attribut `class` :

![image](https://user-images.githubusercontent.com/300823/77330463-b1a4c200-6d1f-11ea-81ff-1b0fc899b8cf.png)

Si c'est OK de se débarrasser de la div englobante (je n'ai pas trouvé de CSS qui dépende de cette classe `certification-results-page`) on peut juste enlever ce `@class="certification-results-page"` qui n'a pas d'effet.

Sinon, si on veut retrouver exactement le même DOM qu'avant le plus simple est de mettre une `<div class="certification-results-page">` englobante dans  `certification-results-page.hbs`.

Reply:
J'ai enlevé le `@class="certification-results-page"` du coup.

Par contre, cette subtilité me perd un peu. On a beaucoup de `@class` dans le code et les classes semblent bien présentes

Reply:
C'est peut-être lié aux glimmer components; ça serait visible pour ce fichier précis du coup parce que c'est l'un des premiers.

A surveiller à l'avenir

Reply:
Oui c'est clairement une nouveauté des composants Glimmer : pas de `div` englobante, les `@arg` sont mis sur `this.args` et du coup pas de traitement automatique de `@class`.